src/pydata_sphinx_theme/__init__.py

```diff
@@ -1084,6 +1085,43 @@ def setup_translators(app):
             app.set_translator(name, translator, override=True)
 
 
+# ------------------------------------------------------------------------------
+# customize method for logo management
+# ------------------------------------------------------------------------------
+
+
+def setup_logo_path(
+    app: Sphinx, pagename: str, templatename: str, context: dict, doctree: nodes.Node
+) -> None:
+    """Set up relative paths to logos (works for all Sphinx versions)."""
+
+    # get informations from the context
+    pathto = context.get("pathto")
+    logo = context.get("logo_url", "") or context.get("logo", "")
+    image_light = context.get("theme_logo", {}).get("image_light")
+    image_dark = context.get("theme_logo", {}).get("image_dark")
+
+    # add the theme_logo to context to make it editable downstream
+    if "theme_logo" not in context:
+        context["theme_logo"] = {}
+
+    # create the logo_url
+    if logo and not isurl(logo):
+        logo = pathto(f"_static/{logo}", resource=True)
+
+    # update light image with the appropriate url
+    if image_light and not isurl(image_light):
+        image_light = pathto(f"_static/{image_light}", resource=True)
+    image_light = image_light or logo
+    context["theme_logo"]["logo_light"] = image_light
+
+    # update dark image with the appropriate url
+    if image_dark and not isurl(image_dark):
+        image_dark = pathto(f"_static/{image_dark}", resource=True)
+    image_dark = image_dark or image_light
+    context["theme_logo"]["logo_dark"] = image_dark
```

Comment:
I just took another look at this and to understand it, I had to re-order the lines a bit and add some comments. Here is where I ended up:

```py
    # get information from the context
    pathto = context.get("pathto")
    logo = context.get("logo_url", "") or context.get("logo", "")

    # make sure theme_logo dictionary exists
    if "theme_logo" not in context:
        context["theme_logo"] = {}

    # get user-specified light/dark logo versions (fallback to None)
    image_light = context["theme_logo"].get("image_light")
    image_dark = context["theme_logo"].get("image_dark")

    # resolve URLs / paths
    if logo and not isurl(logo):
        logo = pathto(f"_static/{logo}", resource=True)
    if image_dark and not isurl(image_dark):
        image_dark = pathto(f"_static/{image_dark}", resource=True)
    if image_light and not isurl(image_light):
        image_light = pathto(f"_static/{image_light}", resource=True)

    # fallback to logo if image_{light|dark} is None
    context["theme_logo"]["logo_light"] = image_light or logo
    context["theme_logo"]["logo_dark"] = image_dark or logo
```

Even after doing this I'm a bit confused as to why there are separate keys `image_light` and `logo_light` (and likewise for `_dark`).  IIUC, `image_light` is set by the user in their `conf.py`, and `logo_light` is then used in the `components/navbar-logo.html` template?  Do they need to be separate keys in order for things to work?  @12rambau I leave it to you to do what you think is best here... maybe more comments is enough? or changing the key names to be more distinct/descriptive? Whatever further improvements you decide to do, kudos for sorting this all out!

Reply:
I'll add more comments because it's effectively quite messy. What I want to acomplish here is set image_light and image_dark based on whatever we found in the conf.py. yes `image_light` is already in the shared "context" passed to each page but as it's shared, I should not modify it.